File: src/main/java/edu/utn/grupo3/reservas/model/Reserva.java
package edu.utn.grupo3.reservas.model;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@NoArgsConstructor(access = AccessLevel.PROTECTED, force = true)
@Table(name = "reservas")
public class Reserva extends ObjetDB {
    private String fecha;
    private String horaDesde;
    private String horaHasta;
    private Integer cantidadReserva;
    private String motivo;
    private String fechaHoraReserva;

    @ManyToOne
    private Solicitante solicitantes;
    @OneToOne
    private Espacio espacios;
    //@ManyToOne
    //private EstadoReserva estadosReserva;

}

File: reservas-frontend/src/app/model/reserva.ts
import { Solicitante } from "./solicitante";
import { Espacio } from "./espacio";

export class Reserva {

    id:number;
    fecha:Date;
    horaDesde:Date;
    horaHasta:Date;
    cantidadReserva:number;
    motivo:String;
    fechaHoraReserva:Date;
    solicitantes:Solicitante;
    espacios:Espacio;

}